object/link.go

```diff
@@ -63,7 +84,11 @@ func (m *MeasuredObject) SetObjectID(id oid.ID) {
 // See also [MeasuredObject.SetObjectID].
 func (m *MeasuredObject) ObjectID() oid.ID {
 	var id oid.ID
-	_ = id.ReadFromV2(m.ID)
+	if m.ID.GetValue() != nil {
+		if err := id.ReadFromV2(m.ID); err != nil {
+			panic(fmt.Errorf("invalid ID: %w", err))
```

Comment:
Here as well. Old behavior was OK to me (especially now with explicitly forbidden zero IDs).

Reply:
easy rule: if user cannot do anything on its part that will lead to an error, and it can only be reached due to an implementation mistakes - all that's left is to panic. Potential dev errors must not be ignored/silenced

> it is easy to lose track of such methods and panic because of a user error

it's always easy for dev to make a mistake, so the concerns and tests should cover that. What is really bad is the loss of the error and the lack of feedback on it to the user. That is, if the developer makes a mistake now, the user will receive code that quietly does not work. This is a fatally wrong approach for me

Reply:
My concern is triggering this panic with external input (we're talking about exchanged structures of various kinds). If this is not possible and we can't have a node panicking upon packet receival (or otherwise getting invalid struct) then OK.

Reply:
understand ur concern. I checked all possible usage patterns of the type and did not find how the user could come to this error. If (if) panic will occur, then we encounter bug leading to data corruption/loss. In this case, panic wont make us happy, but at least it will stop the malicious software